pkg/rabbit/connections_handler.go

```diff
@@ -119,23 +118,35 @@ func (r *RabbitMQHelper) SetupRabbitMQConnectionAndChannel(
 	configFunction func(RabbitMQConnectionInterface, RabbitMQChannelInterface) error,
 	dialFunc func(string) (RabbitMQConnectionWrapperInterface, error)) {
 	var err error
-	retryConnection, retryChannel := true, true
-	for retryConnection || retryChannel {
+	for {
 		// Wait one cycle duration always except the first time
 		if !r.firstSetup {
 			time.Sleep(time.Millisecond * r.cycleDuration)
```

Comment:
If you move the sleep to L146, can remove the whole `firstSetup` tracking

Reply:
No cause in the case that the connection channel and config always succeed (but still somethings goes wrong when starting to use them, for example inspecting for a queue) it can get in an infinite loop of success and no wait

Reply:
We shouldn't be recreating connections because a failure happened elsewhere with a call to rabbtimq.

Reply:
In the case of the queue inspect the connection and channel creates fine, but if the queueInspect fails it close the channel (weird design decision but wel...) so its a loop where everything Is working except from the queue and we have to constantly recreate stuff